lib/puppet/type/cisco_route_map.rb

```diff
@@ -0,0 +1,1642 @@
+
+  newproperty(:match_route_type_intra_area) do
+    desc 'Enable match OSPF intra area route'
+
+    newvalues(:true, :false, :default)
+  end # property match_route_type_intra_area
+
+  newproperty(:match_route_type_level_1) do
+    desc 'Enable match IS-IS level-1 route'
+
+    newvalues(:true, :false, :default)
+  end # property match_route_type_level_1
+
+  newproperty(:match_route_type_level_2) do
+    desc 'Enable match IS-IS level-2 route'
+
+    newvalues(:true, :false, :default)
+  end # property match_route_type_level_2
+
+  newproperty(:match_route_type_local) do
+    desc 'Enable match locally generated route'
+
+    newvalues(:true, :false, :default)
+  end # property match_route_type_local
+
+  newproperty(:match_route_type_nssa_external) do
+    desc 'Enable match nssa-external route (OSPF type 1/2)'
+
+    newvalues(:true, :false, :default)
+  end # property match_route_type_nssa_external
+
+  newproperty(:match_route_type_type_1) do
+    desc 'Enable match OSPF external type 1 route'
+
+    newvalues(:true, :false, :default)
+  end # property match_route_type_type_1
+
+  newproperty(:match_route_type_type_2) do
+    desc 'Enable match OSPF external type 2 route'
+
+    newvalues(:true, :false, :default)
+  end # property match_route_type_type_2
+
+  newproperty(:match_src_proto, array_matching: :all) do
+    format = '[pr1, pr2]'
+    desc 'Match source protocol. An array of [pr1, pr2 and so on]' \
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
+      value.inspect
+    end
+
+    def is_to_s(value)
+      value.inspect
+    end
+
+    munge do |value|
+      begin
+        return value = :default if value == 'default'
+        value
+      end
+    end
+  end # property match_src_proto
+
+  newproperty(:match_tag, array_matching: :all) do
+    format = '[tag1, tag2]'
+    desc 'Match tag of route. An array of [tag1, tag2 and so on]' \
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
+      value.inspect
+    end
+
+    def is_to_s(value)
+      value.inspect
+    end
+
+    munge do |value|
+      begin
+        return value = :default if value == 'default'
+        value
+      end
+    end
+  end # property match_tag
+
+  newproperty(:match_vlan) do
+    desc "Match Vlan ID. Valid values are string,
+          keyword 'default'"
+
+    munge do |value|
```

Comment:
We have some prior art for processing vlan ranges in cisco_vlan.  For consistency, we might want to follow the same convention.

https://github.com/cisco/cisco-network-puppet-module/blob/develop/lib/puppet/type/cisco_vlan.rb#L174